helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -108,17 +107,14 @@ private void refresh(ClusterConfig clusterConfig, InstanceConfig instanceConfig,
    * @param assignableReplica - the replica to be assigned
    */
   void assign(AssignableReplica assignableReplica) {
-    if (!addToAssignmentRecord(assignableReplica, _currentAssignments)) {
+    if (addToAssignmentRecord(assignableReplica)) {
+      assignableReplica.getCapacity().entrySet().stream().forEach(
+          capacity -> updateCapacityAndUtilization(capacity.getKey(), capacity.getValue()));
+    } else {
```

Comment:
Why don't we just throw this exception directly inside `addToAssignmentRecord()`? That way code is more concise. No need for this if statement. Also, we avoid having to return a boolean in `addToAssignmentRecord()`, which, to me at least, is quite unreadable...

Reply:
Good idea. Will change.